marshal/claim.go

```diff
@@ -261,6 +263,13 @@ func (c *claim) teardown(releasePartition bool) bool {
 	// is reasonable. Held because of using offsetCurrent below.
 	c.lock.RLock()
 	defer c.lock.RUnlock()
+
+	// Advise the consumer that this claim is terminating, this is so that the consumer
+	// can release other claims if we've lost part of a topic
+	if c.consumer != nil {
+		go c.consumer.claimTerminated(c, releasePartition)
```

Comment:
I think my point was mainly around when do I really know that the offsets have been committed so I can terminate the process? Doesn't this break that?